### ISSUE TITLE
Inquiry About Pretrained Models for Reproducing Your Work

### ISSUE THREAD

**Issue by wangyp33** [glin2022/atop#5]
Dear author，I recently read your paper and found your work on AP particularly insightful.

I’m currently attempting to reproduce the experiments, but encountered difficulties obtaining the pretrained classifier models (e.g., ./models/state_dicts/resnet18_cifar10.pth) mentioned in the paper. Could you kindly clarify:

- Whether these models are publicly available?  (e.g., official repository, third-party platform)?

- If not, would it be possible to share the pretrained weights or provide training details (e.g., hyperparameters, datasets)?

Any guidance would be greatly appreciated. Thank you for your time and contribution to the community!

**Comment by wangyp33** [comment]
I tried adapting third-party resnet18_cifar10.pth (e.g., Hugging Face's) and encountered the following error when loading weights:

Error(s) in loading state_dict for ResNet: Missing key(s) in state_dict: .....      Unexpected key(s) in state_dict:....

**Comment by glin2022** [comment]
Thanks for your interest in our work!

In a previous issue (https://github.com/glin2022/atop/issues/3), someone also requested the ResNet checkpoint. However, we noticed that the output differs from the original log. I realized that the original ResNet used for training may have been unintentionally overwritten. Therefore, I have provided the full model checkpoint (https://drive.google.com/drive/folders/17qP_PWL1VXTnrZoRWhaESjvcIqkggArg?usp=share_link).

For reference, the classifier used for AToP is a standard ResNet18. Additionally, you can use other classifiers as well, as this will not significantly affect the performance. However, to facilitate the training of the purifier (GAN) model, we modified the input size of ResNet18 to match the output size of the GAN, rather than the original size 32×32.

**Comment by wangyp33** [comment]
Sorry to bother you, I'm not sure what full model checkpoint means :
states_RT1_adv.pth  states_RT1_cln.pth  states_RT2_3_adv.pth   states_RT2_3_cln.pth

- how to use the checkpoint you provided?

maybe "***_adv.pth " denote purifier model, while "***_cln.pth " denote classifier model? What dataset were they trained on, and what classifiers and purifiers were used

**Comment by glin2022** [comment]
Sorry for the late reply.

These checkpoints correspond to Table 7 in the paper. They include models trained with adversarial loss (_adv.pth) and without adversarial loss (_cln.pth)  after the RT process. Since RT-3 is a repeated application of RT-2, they share the same model.
All results were trained and tested on CIFAR-10.

For reference, our method is independent of specific purifiers (generators) and classifiers. The core idea is to **concatenate the purifier and classifier**, then fine-tune the parameters using adversarial loss. (Note that we optimize the model by **freezing the classifier parameters and only updating the purifier parameters.**)

**Comment by wangyp33** [comment]
👍 Thank you for your response and code sharing. I think your work highly worthy of following and in-depth research.
